test/contracts/LSP.spec.ts

```diff
@@ -293,4 +299,73 @@ describe("LSP", function () {
     const expectedCollateralBalance = oldCollateralBalance.add(tokensToRedeem)
     expect(newCollateralBalance).toEqBN(expectedCollateralBalance)
   })
+  it("Should change state to ExpiredPriceRequested after expire", async function () {
+    // Check that sponsor still has enough L / S Tokens
+    const syntheticBalances = await contracts.LSP.getPositionTokens(
+      namedAccounts.deployer
+    )
+    expect(syntheticBalances[0]).toBeGteBN(tokensToSettle)
+    expect(syntheticBalances[1]).toBeGteBN(tokensToSettle)
+
+    // Check contract state
+    const ContractState = {
+      Open: 0,
+      ExpiredPriceRequested: 1,
+      ExpiredPriceReceived: 2,
+    }
+
+    let contractState: number = await contracts.LSP.contractState()
+
+    expect(contractState).toEqual(ContractState.Open)
+
+    // fast forward 350 seconds
+    await ethers.provider.send("evm_increaseTime", [350])
```

Comment:
Didn't you say this is one of the most complex issues in smart contract testing? Looks pretty straight-forward to me 😉